catroid/src/main/java/org/catrobat/catroid/ProjectManager.java

```diff
@@ -405,6 +405,37 @@ public boolean initializeDefaultProject(Context context) {
 		}
 	}
 
+	public static Project mergeProjects(String mergedProjectName, Project project1, Project project2) {
+		Project mergedProject = new Project();
+		mergedProject.setXmlHeader(project1.getXmlHeader());
+		mergedProject.getSettings().addAll(project1.getSettings());
+
+		mergedProject.getUserVariables().addAll(project1.getUserVariables());
+		mergedProject.getUserVariables().addAll(project2.getUserVariables());
+
+		mergedProject.getUserLists().addAll(project1.getUserLists());
+		mergedProject.getUserLists().addAll(project2.getUserLists());
+
+		mergedProject.setName(mergedProjectName);
+
+		for (Scene scene : project1.getSceneList()) {
+			for (Scene checkScene : project2.getSceneList()) {
```

Comment:
please rename "scene" and "checkScene" to "sceneInProject1" and "sceneInProject2".
it was unnecessarily hard to comprehend whats going on there and what "checkScene" means.

Reply:
i now found that there already is a getSceneNames method in Project.
so a 
`for (String sceneName : project1.getSceneNames()) {
    if (project2.getSceneNames().contains(sceneName) { return null }
}`

and if that succeeds 
`mergedProject.getSceneList().addAll(project1.getSceneList())
mergedProject.getSceneList().addAll(project2.getSceneList())
`

Reply:
however, all of this is going to leave you with a broken project since none of the files (sounds, looks) are moved, and the sprites inside the scenes that contain references to those are going to be broken.
also care must be taken before doing any changes, like updating the paths, etc since the scenes, sprites, variables, etc are references that coming from the original projects.
so if youd update the filename in any sprite and then somewhere project1 is saved, it will break it.